helix-lock/src/main/java/org/apache/helix/lock/LockInfo.java

```diff
@@ -45,54 +40,105 @@
    * Initialize a default LockInfo instance
    */
   private LockInfo() {
     _record = new ZNRecord(ZNODE_ID);
-    setLockInfoFields(DEFAULT_OWNER_TEXT, DEFAULT_MESSAGE_TEXT, DEFAULT_TIMEOUT_LONG);
+    setLockInfoFields(LockConstants.DEFAULT_USER_ID, LockConstants.DEFAULT_MESSAGE_TEXT,
+        LockConstants.DEFAULT_TIMEOUT_LONG, LockConstants.DEFAULT_PRIORITY_INT,
+        LockConstants.DEFAULT_WAITING_TIMEOUT_LONG, LockConstants.DEFAULT_CLEANUP_TIMEOUT_LONG,
+        LockConstants.DEFAULT_USER_ID, LockConstants.DEFAULT_PRIORITY_INT,
+        LockConstants.DEFAULT_WAITING_TIMEOUT_LONG,
+        LockConstants.DEFAULT_REQUESTING_TIMESTAMP_LONG);
   }
 
   /**
    * Initialize a LockInfo with a ZNRecord, set all info fields to default data
    * @param znRecord The ZNRecord contains lock node data that used to initialize the LockInfo
    */
   public LockInfo(ZNRecord znRecord) {
-    this();
-    if (znRecord != null) {
-      String ownerId = znRecord.getSimpleField(LockInfoAttribute.OWNER.name());
-      String message = znRecord.getSimpleField(LockInfoAttribute.MESSAGE.name());
-      long timeout = znRecord.getLongField(LockInfoAttribute.TIMEOUT.name(), DEFAULT_TIMEOUT_LONG);
-      setLockInfoFields(ownerId, message, timeout);
+    _record = new ZNRecord(ZNODE_ID);
+    if (znRecord == null) {
+      znRecord = new ZNRecord(ZNODE_ID);
     }
+    String ownerId = znRecord.getSimpleField(LockInfoAttribute.OWNER.name()) == null
+        ? LockConstants.DEFAULT_USER_ID : znRecord.getSimpleField(LockInfoAttribute.OWNER.name());
+    String message = znRecord.getSimpleField(LockInfoAttribute.MESSAGE.name()) == null
+        ? LockConstants.DEFAULT_MESSAGE_TEXT
+        : znRecord.getSimpleField(LockInfoAttribute.MESSAGE.name());
+    long timeout =
+        znRecord.getLongField(LockInfoAttribute.TIMEOUT.name(), LockConstants.DEFAULT_TIMEOUT_LONG);
+    int priority =
+        znRecord.getIntField(LockInfoAttribute.PRIORITY.name(), LockConstants.DEFAULT_PRIORITY_INT);
+      long waitingTimeout = znRecord.getLongField(LockInfoAttribute.WAITING_TIMEOUT.name(),
+          LockConstants.DEFAULT_WAITING_TIMEOUT_LONG);
+      long cleanupTimeout = znRecord.getLongField(LockInfoAttribute.CLEANUP_TIMEOUT.name(),
+          LockConstants.DEFAULT_CLEANUP_TIMEOUT_LONG);
+      String requestorId = znRecord.getSimpleField(LockInfoAttribute.REQUESTOR_ID.name());
+      int requestorPriority = znRecord.getIntField(LockInfoAttribute.REQUESTOR_PRIORITY.name(),
+          LockConstants.DEFAULT_PRIORITY_INT);
+      long requestorWaitingTimeout = znRecord
+          .getLongField(LockInfoAttribute.REQUESTOR_WAITING_TIMEOUT.name(),
+              LockConstants.DEFAULT_WAITING_TIMEOUT_LONG);
+      long requestingTimestamp = znRecord
+          .getLongField(LockInfoAttribute.REQUESTING_TIMESTAMP.name(),
+              LockConstants.DEFAULT_REQUESTING_TIMESTAMP_LONG);
+      setLockInfoFields(ownerId, message, timeout, priority, waitingTimeout, cleanupTimeout,
```

Comment:
nit, combine this with line 71, then we can call the constructor directly instead of 2 lines.